openverse_catalog/dags/providers/provider_api_scripts/brooklyn_museum.py

```diff
@@ -2,186 +2,155 @@
-        license_url = _get_license_url(rights_info)
-        logger.debug(license_url)
-        if license_url is not None:
-            id_ = object_.get("id", "")
-            complete_object_data = _get_object_json(endpoint=ENDPOINT + str(id_))
-            if complete_object_data is None:
-                continue
-            _handle_object_data(data=complete_object_data, license_url=license_url)
-
-
-def _handle_object_data(data, license_url):
-    image_info = data.get("images")
-    if image_info is not None:
-        id_ = data.get("id", "")
-        title = data.get("title", "")
-        foreign_url = f"https://www.brooklynmuseum.org/opencollection/objects/{id_}"
-        metadata = _get_metadata(data)
-        creators = _get_creators(data)
-
-        for image in image_info:
-            foreign_id = image.get("id", "")
-            image_url = _get_image_url(image)
-            if image_url is None:
-                continue
-            height, width = _get_image_sizes(image)
-            license_info = get_license_info(license_url=license_url)
-            image_store.add_item(
-                foreign_landing_url=foreign_url,
-                image_url=image_url,
-                license_info=license_info,
-                foreign_identifier=foreign_id,
-                width=width,
-                height=height,
-                title=title,
-                meta_data=metadata,
-                creator=creators,
+            offset = prev_query_params["offset"] + self.batch_limit
+        return {**self._common_args, "offset": offset}
+
+    @staticmethod
+    def _get_data_from_response(response_json) -> dict | None:
+        if response_json and response_json.get("message", "").lower() == "success.":
+            return response_json.get("data")
+        return None
+
+    def get_batch_data(self, response_json) -> list | None:
+        return self._get_data_from_response(response_json)
+
+    @staticmethod
+    def _get_license_url(rights_info):
+        elements = html.fromstring(rights_info.get("description", ""))
+        cc_links = [
+            link
+            for _, _, link, _ in elements.iterlinks()
+            if "https://creativecommons.org/" in link
+        ]
+        license_url = cc_links[0] if len(cc_links) == 1 else None
+        return license_url
+
+    @staticmethod
+    def _get_image_sizes(image):
+        height, width = None, None
+        size_list = image.get("derivatives", "")
+        if isinstance(size_list, list):
+            # Pull out the largest sized image we have access to
+            # https://www.brooklynmuseum.org/api/new-documentation/#definition-Image
+            size_type = image.get("largest_derivative", "")
+            for size in size_list:
+                if size.get("size", "") == size_type:
+                    height = size.get("height")
+                    width = size.get("width")
+        return height, width
+
+    @staticmethod
+    def _get_metadata(data):
```

Comment:
I think we should clean up dictionaries like this to not add values that are None by running something like
```python
return { k: v for k, v in meta_data.items() if v is not None}
```